interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -104,6 +105,19 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
       case _ => PartTypeError.invalidNel
     }
 
+    def withTwoChildrenOfType[A: universe.TypeTag, R: universe.TypeTag](op: (A, A) => R) = {
+      val expectedType = Typed.fromDetailedType[A]
+      val resultType = Typed.fromDetailedType[R]
+      withTypedChildren {
+        case lst@left :: right :: Nil if lst.forall(_.typingResult.canBeSubclassOf(expectedType)) =>
+          val res = left.typingResult.value
+            .flatMap(l => right.typingResult.value.map((l, _)))
+            .map{ case (x, y) => op(x.asInstanceOf[A], y.asInstanceOf[A]) }
```

Comment:
I think that we could create some generic extractor which could ensure that if `typingResult.canBeSubclassOf` some type then `typingResult.value` will return `Option[T]`. WDYT?

Reply:
Do you have some reasonable situation where `typingResult.value.map(_.asInstanceOf[T])` fails, but `typingResult.valueAsType[T]` works correctly? I think that, every time .value is used, we already are sure what type we expect there, and it would be better to fail than try to handle such case.

Reply:
I'm not sure if I get you correct. By my comment I mean sth like this: https://github.com/TouK/nussknacker/pull/3442/files  . Do you think it gives some value? The problem I've seen was that between `canBeSubclassOf` and `asInstanceOf` can happens many things which cause ClassCastException at the end. Thanks to extractors, it is done in one step so we are safe that nothing danger happen. It is similar approach to: https://www.scala-exercises.org/shapeless/type_safe_cast

Reply:
I thought you ment something different. What you proposed is good, I have merged it.